test/e2e_node/container_lifecycle_test.go

```diff
@@ -537,6 +537,228 @@ var _ = SIGDescribe(framework.WithNodeConformance(), "Containers Lifecycle", fun
 		framework.ExpectNoError(results.HasNotRestarted(init3))
 		// while the regular container did
 		framework.ExpectNoError(results.HasRestarted(regular1))
+
+		ginkgo.It("should not restart init container when updated with a new image", func() {
+			init1 := "init-1"
+			regular1 := "regular-1"
+
+			podSpec := &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "initcontainer-update-img-always",
+				},
+				Spec: v1.PodSpec{
+					RestartPolicy: v1.RestartPolicyAlways,
+					InitContainers: []v1.Container{
+						{
+							Name:  init1,
+							Image: busyboxImage,
+							Command: ExecCommand(init1, execCommand{
+								Delay:    1,
+								ExitCode: 0,
+							}),
+						},
+					},
+					Containers: []v1.Container{
+						{
+							Name:  regular1,
+							Image: busyboxImage,
+							Command: ExecCommand(regular1, execCommand{
+								Delay:    10,
+								ExitCode: -1,
+							}),
+						},
+					},
+				},
+			}
+			preparePod(podSpec)
+
+			client := e2epod.NewPodClient(f)
+			podSpec = client.Create(context.TODO(), podSpec)
+
+			ginkgo.By("running the pod", func() {
+				err := e2epod.WaitForPodNameRunningInNamespace(context.TODO(), f.ClientSet, podSpec.Name, podSpec.Namespace)
+				framework.ExpectNoError(err)
+			})
+
+			ginkgo.By("updating the image", func() {
+				client.Update(context.TODO(), podSpec.Name, func(pod *v1.Pod) {
+					pod.Spec.Containers[0].Image = imageutils.GetE2EImage(imageutils.Nginx)
+				})
+
+			})
+
+			ginkgo.By("verifying that the containers do not restart", func() {
+				err := e2epod.WaitForPodSuccessInNamespace(context.TODO(), f.ClientSet, podSpec.Name, podSpec.Namespace)
+				framework.ExpectNoError(err)
+				podSpec, err = client.Get(context.TODO(), podSpec.Name, metav1.GetOptions{})
+				framework.ExpectNoError(err)
+				results := parseOutput(context.TODO(), f, podSpec)
```

Comment:
```suggestion
				results := parseOutput(ctx, f, podSpec)
```

Reply:
Other test cases would also benefit from reusing test context instead of creating their own contexts.

Reply:
good note, will fix this in the other tests in a separate PR